### PR TITLE
Fix Failing Dailies

### PR DESCRIPTION
**Pull request by pauladkisson**
Dailies were failing bc os-versions and python-versions were not being propagated properly.

Now fixed -- see here: https://github.com/catalystneuro/neuroconv/actions/runs/11463322663

Daily neuroconv docker tests are failing (see #1121), but since that is a separate problem, we can solve it separately.

### CODE REVIEW

File: .github/workflows/neuroconv_docker_testing.yml [hunk 1]
```diff
@@ -81,7 +81,7 @@ jobs:
 
       - name: Pull docker image
         run: |
-          docker pull ghcr.io/catalystneuro/neuroconv:latest
+          docker pull ghcr.io/catalystneuro/neuroconv:v0.6.4
```
Comment (h-mayorquin) [inline .github/workflows/neuroconv_docker_testing.yml:84]: Can you add more context about this change?

**Review by h-mayorquin — COMMENTED** [review]
What do you think about changing the following lines to make the argument non-required for this and the other workflows:

https://github.com/catalystneuro/neuroconv/blob/977144edbd2d0412b7a47fbc52e916376311d579/.github/workflows/testing.yml#L8-L11

That way we only need to keep those default versions in one place.

**Comment by pauladkisson** [comment]
> That way we only need to keep those default versions in one place.

Yeah, I think that's a good idea.

**Comment by h-mayorquin** [comment]
> > That way we only need to keep those default versions in one place.
> 
> Yeah, I think that's a good idea.

See here for more related discussion:

https://github.com/catalystneuro/neuroconv/pull/1117#issuecomment-2427842819

Tag me when this is ready to take a look.

**Comment by pauladkisson** [comment]
@h-mayorquin, this is ready to go.

**Comment by h-mayorquin** [comment]
Where are you setting `vars.ALL_PYTHON_VERSIONS`? I would like to have the ability to open a PR when adding support for new versions as I would like to run the testing suit before merging.

**Comment by h-mayorquin** [comment]
I would prefer to have them in code so we can have a workflow where changing them requires and triggers a PR.

**Review by bendichter — APPROVED** [review]

**Comment by bendichter** [comment]
Oh yeah, that's a good point. Is there a way to put this as a variable in code?

**Comment by pauladkisson** [comment]
> Oh yeah, that's a good point. Is there a way to put this as a variable in code?

As far as I could tell, the only way to include this in the code is to define it separately for each workflow. If we want all the workflows to reference a single variable, I think it has to be a github repository variable: https://github.com/catalystneuro/neuroconv/settings/variables/actions

**Comment by h-mayorquin** [comment]
We can either use the defaults to pass to reduce the duplication (simpler and probably the way to go) or you can write the variable to a file and use bash or an action to read them.

**Comment by pauladkisson** [comment]
@h-mayorquin how about this solution?

**Comment by h-mayorquin** [comment]
Yes, I am glad that the reading from a file option worked. This LGTM.

**Pull request merged**